### PR TITLE
Chapter context

### LINKED ISSUE #55: [anheumai-dparys] Context
Complete the Context lesson

### PR DESCRIPTION
**Pull request by dariuszparys**
Interesting chapter, good introduction to context in go and how to
use them.

Closes #55

### CODE REVIEW

File: anheumai-dparys/context/context_test.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+func (s *SpyStore) Fetch(ctx context.Context) (string, error) {
+	data := make(chan string, 1)
+
+	go func() {
+		var result string
+		for _, c := range s.response {
+			select {
+			case <-ctx.Done():
+				s.t.Log("they cancelled why, why???")
```
Comment (oliverscheer) [inline anheumai-dparys/context/context_test.go:25]: Why? Who? What? The big questions of the universe. Hopefully they will be answered one day. But not today :-)

File: anheumai-dparys/context/context_test.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	response string
+	t        *testing.T
+}
+
+func (s *SpyStore) Fetch(ctx context.Context) (string, error) {
+	data := make(chan string, 1)
+
+	go func() {
+		var result string
+		for _, c := range s.response {
```
Comment (oliverscheer) [inline anheumai-dparys/context/context_test.go:22]: What is c? a single context? I haven't done that chapter yet.

**Review by oliverscheer — APPROVED** [review]

File: anheumai-dparys/context/context_test.go (hunk 2 above)
Comment (dariuszparys) [reply anheumai-dparys/context/context_test.go:22]: no, we're iterating here over a string character by character, c == character

Comment (oliverscheer) [reply anheumai-dparys/context/context_test.go:22]: Thank you very much for clarification. I really appreciate this :-)

**Pull request closed without merging**